### PR TITLE
Fix preload example to avoid excessive preloading

### LINKED ISSUE #5350: Issue with "Link types: preload": Misleading, Possibly Incorrect Example
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload

#### What information was incorrect, unhelpful, or incomplete?

The "[Including a MIME Type](https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload#including_a_mime_type)" section is misleading or incorrect.

The section provides an example in which two video sources of different media types have rel="preload" links in the head. The section text states that browsers that don't support the MP4 media type will fall back to the WebM source _and_ lose the benefits of preloading. If this is true, it is unclear why a rel="preload" link would be given in the head for the WebM media type, and the presence of this link lends itself to misunderstanding of type attribute on links. If preload benefits would not be lost in this example for browsers that support WebM but not MP4, the section text is incorrect.

#### Specific section or headline?

https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload#including_a_mime_type

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/link_types/preload`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preload
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/link_types/preload/index.html
* Last commit: https://github.com/mdn/content/commit/788d0bae21eea799302841d3b38ae34ec0c355e3
* Document last modified: 2021-05-20T18:18:07.000Z

</details>

### PR DESCRIPTION
**Pull request by sideshowbarker**
Fixes https://github.com/mdn/content/issues/5350

---

This change drops an unnecessary preload from a code example, and changes the text after the example to read as follows:

> <p>The code in the example above causes the <code>video/mp4</code> video to be preloaded — and for users who have <code>video/mp4</code> support in their browsers, causes the <code>video/mp4</code> video to actually be used (since it’s the first <a href="/en-US/docs/Web/HTML/Element/source"><code>&lt;source&gt;</code></a> specified). That makes the video player hopefully smoother/more responsive for users who have <code>video/mp4</code> support in their browsers.</p>
> 
> <p>Note that for users whose browsers have both <code>video/mp4</code> and <code>video/webm</code> support, if in that code a <code>&lt;link rel="preload" href="sintel-short.webm" as="video" type="video/webm"&gt;</code> element were also specified, then <em>both</em> the <code>video/mp4</code> and <code>video/webm</code> videos would be preloaded — even though only one of them would actually be used.</p>
> 
> <p>Therefore, it’s not a best practice to specify preloading for multiple types of the same resource in the case where the majority of your users have browsers that support all the types. Instead, the best practice is to specify preloading only for the type the majority of your users are likely to actually use. That’s why the code in the example above doesn’t specify preloading for the <code>video/webm</code> video.</p>
> 
> <p>However, the lack of preloading doesn’t prevent the <code>video/webm</code> video from actually being used by those who need it: for users whose browsers don’t have <code>video/mp4</code> support but do have <code>video/webm</code> support, the code in the example above does still cause the <code>video/webm</code> video to be used — but it does so without also causing it to also be preloaded unnecessarily for the majority of other users.</p>

@yoavweiss Your review on the wording here would be much appreciated.

### CODE REVIEW

File: files/en-us/web/html/link_types/preload/index.html [hunk 1]
```diff
@@ -113,7 +112,13 @@ <h2 id="Including_a_MIME_type">Including a MIME type</h2>
   &lt;/video&gt;
 &lt;/body&gt;</pre>
 
-<p>So in this case, browsers that support MP4s will preload and use the MP4, making the video player hopefully smoother/more responsive for users. Browsers that don't support MP4 can still load the WebM version, but don't get the advantages of preloading. This shows how preloading content can be combined with the philosophy of progressive enhancement.</p>
+<p>The code in the example above causes the <code>video/mp4</code> video to be preloaded — and for users who have <code>video/mp4</code> support in their browsers, causes the <code>video/mp4</code> video to actually be used (since it’s the first {{htmlelement("source")}} specified). That makes the video player hopefully smoother/more responsive for users who have <code>video/mp4</code> support in their browsers.</p>
```
Comment (yoavweiss) [inline files/en-us/web/html/link_types/preload/index.html:115]: s/video to be preloaded/video to be preloaded only in supporting browsers/

File: files/en-us/web/html/link_types/preload/index.html [hunk 2]
```diff
@@ -113,7 +112,13 @@ <h2 id="Including_a_MIME_type">Including a MIME type</h2>
...
+
+<p>Note that for users whose browsers have both <code>video/mp4</code> and <code>video/webm</code> support, if in that code a <code>&lt;link rel="preload" href="sintel-short.webm" as="video" type="video/webm"&gt;</code> element were also specified, then <em>both</em> the <code>video/mp4</code> and <code>video/webm</code> videos would be preloaded — even though only one of them would actually be used.</p>
+
+<p>Therefore, it’s not a best practice to specify preloading for multiple types of the same resource in the case where the majority of your users have browsers that support all the types. Instead, the best practice is to specify preloading only for the type the majority of your users are likely to actually use. That’s why the code in the example above doesn’t specify preloading for the <code>video/webm</code> video.</p>
```
Comment (yoavweiss) [inline files/en-us/web/html/link_types/preload/index.html:119]: s/not a best practice/discouraged/ ?

Comment (yoavweiss) [reply files/en-us/web/html/link_types/preload/index.html:119]: I'd also remove the "in the case" caveat. I don't think it's a good idea to preload multiple types as long as we don't have a mechanism that would prevent multiple downloads.

**Comment by yoavweiss** [comment]
Something I just realized - I think that in terms of implementations video/audio preload is not widely supported (I vaguely rememeber support removed from WebKit and maybe Chromium). Worthwhile to verify, although it may make sense to land this in the interim.

File: files/en-us/web/html/link_types/preload/index.html (hunk 1 above)
Comment (sideshowbarker) [reply files/en-us/web/html/link_types/preload/index.html]: > s/video to be preloaded/video to be preloaded only in supporting browsers/

OK, thanks — fixed

File: files/en-us/web/html/link_types/preload/index.html (hunk 2 above)
Comment (sideshowbarker) [reply files/en-us/web/html/link_types/preload/index.html]: > s/not a best practice/discouraged/ ?

Yes, thanks — changed to that

> I'd also remove the "in the case" caveat. I don't think it's a good idea to preload multiple types as long as we don't have a mechanism that would prevent multiple downloads.

Thanks, yup, makes sense — changed to that

**Comment by sideshowbarker** [comment]
> Something I just realized - I think that in terms of implementations video/audio preload is not widely supported (I vaguely remember support removed from WebKit and maybe Chromium).

Well yeah, that’s a good point — the article already mentions this near the beginning:

> Note: video preloading is included in the Preload spec, but is not currently implemented by browsers.

…so given that, I’m not sure how we ended up we ended up choosing to show video preloading as the basis for this example…

> Worthwhile to verify

Yeah, I just now doublechecked in Safari Tech Preview, Firefox Nightly, and Canary, and the example code didn’t cause preloading in any of them (instead it just causes console warnings).

> although it may make sense to land this in the interim.

Yeah — ideally I’d rather we just replaced the entire example. Maybe somebody else will come along later who can make the time to do that. But for now, I’ve at least added more warning language —

> <p>You can see an example of this in our video example (see the <a href="https://github.com/mdn/html-examples/tree/master/link-rel-preload/video">full source code</a>, and also <a href="https://mdn.github.io/html-examples/link-rel-preload/video/">the live version</a>), a code snippet from which is shown below. <b>And while this code won’t actually cause preloading in any browsers — because video preloading isn’t yet supported in any browsers — it still illustrates the core behavior behind preloading in general.</b></p>

**Review by yoavweiss — APPROVED** [review]
LGTM

**Comment by sideshowbarker** [comment]
> LGTM

Cool @yoavweiss Thanks much for the review and suggestions for refinements

**Review by rachelandrew — APPROVED** [review]
lgtm :)

**Pull request merged**